### PR TITLE
[RW-6995][risk=low] DeleteRuntime support deleting PD or keep PD

### PR DESCRIPTION
**Pull request by YintongMa**
In this PR, we extend the deleteRuntime API to support deleting PD or keep PD if the runtime is using PD. Default to false if not specified.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/RuntimeController.java
Comment (yonghaoy) [inline api/src/main/java/org/pmiops/workbench/api/RuntimeController.java:321]: deleteDisk will be null if you merge this PR, is that okay if we just pass null to Leo? 
Suggest to check Null, and pass false if that is null

File: api/src/main/java/org/pmiops/workbench/api/RuntimeController.java
Comment (yonghaoy) [inline api/src/main/java/org/pmiops/workbench/api/RuntimeController.java:315]: I think this will break UI if you did not set that in UI. Because UI is passing null now:
https://github.com/all-of-us/workbench/blob/master/ui/src/app/utils/runtime-utils.tsx#L303

My previous comment is suggesting use `false` if the value is null request. Like:
`deleteDisk == null? false: deletDisk`

Or you can change UI along with the same PR, but I am note sure how many places you need to change, it would be just that line.

File: api/src/main/java/org/pmiops/workbench/api/RuntimeController.java
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/api/RuntimeController.java:325]: I think you miss the 2nd part "and pass false if that is null", not fail the request

File: api/src/main/java/org/pmiops/workbench/api/RuntimeController.java
Comment (YintongMa) [reply api/src/main/java/org/pmiops/workbench/api/RuntimeController.java]: That makes sense! I prefer the first option. It's more complete for the backend api itself.

**Review by yonghaoy — COMMENTED** [review]
I think the UI faliure is related with your change. 
You may change 
https://github.com/all-of-us/workbench/blob/master/ui/src/app/utils/leo-runtime-initializer.tsx#L227
to 
`await runtimeApi().deleteRuntime(this.workspaceNamespace, false, {signal: this.pollAbortSignal});`
But I have very limited knowledge on UI defer to
@calbach

File: api/src/main/java/org/pmiops/workbench/api/RuntimeController.java
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/api/RuntimeController.java:323]: nit: another way of writing this which might make the default value slightly more clear: `Optional.ofNullable(deleteDisk).orElse(false)`

**Review by calbach — COMMENTED** [review]
Yep, this swagger change will result in updates to the typescript fetch clients, per those compilation errors. You'll need to update that location, as well as runtime-utils.tsx.

**Comment by yonghaoy** [comment]
> Yep, this swagger change will result in updates to the typescript fetch clients, per those compilation errors. You'll need to update that location, as well as runtime-utils.tsx.

@YintongMa See https://github.com/all-of-us/workbench/tree/master/ui#all-of-us-workbench-ui
To generate swagger: run `yarn codegen`

File: api/src/main/java/org/pmiops/workbench/notebooks/LeonardoNotebooksClient.java
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/notebooks/LeonardoNotebooksClient.java:37]: nit: would prefer that this takes a primitive boolean instead so that the Java interface is as simple as possible. Looks like the calling code already forces it to be non-null, which is great.

**Review by calbach — APPROVED** [review]

**Review by yonghaoy — APPROVED** [review]

File: api/src/main/java/org/pmiops/workbench/notebooks/LeonardoNotebooksClient.java
Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/notebooks/LeonardoNotebooksClient.java:37]: ^ FYI: you may have missed this comment

**Pull request merged**
